packages/backend/src/studio-api-impl.ts

```diff
@@ -41,31 +85,31 @@ export class StudioApiImpl implements StudioAPI {
   }
 
   async getCategories(): Promise<Category[]> {
-    return content.categories;
+    return this.catalog.categories;
   }
```

Comment:
not sure why it's a Promise ?

Reply:
(same remark for almost all methods)

as catalog is a model, I'm not sure about the usage of async/promises for reading from it

Reply:
I think they are async to respect the interface (class) defined in `shared/src/StudioAPI.ts`, where they are returning a Promise, and from the client side, they are really returning a Promise, due to the wrapper sending a message.

Reply:
ok got it.

I thought that client was getting the model object asynchronously but then it was just fields

client:
```ts
const catalog = await rpc.get(CatalogAPI.class).getCatalog();
const categories = catalog.categories;
const something = catalog.something;
const else = catalog.else;
```

and not a call each time we need something like
```ts
const categories = await rpc....getCategories();
const something = await rpc....getSomething();
const else = await rpc....getElse();

Reply:
What I expect to do with the next PR is to have the backend (this class) post a message to the frontend with the new content of the catalog every time it changes (`in newCatalog()`). On the frontend, I wonder what would be the best approach: having a single `catalog` store, or 3 (or more) stores: `recipes`, `models`, `categories`.

Or, even, having 3 different messages (newRecipes, newModels, newCategories)? But inihs case we would have to check which ones have been modified, to push only the modified ones.

WDYT?